lib/modules/manager/maven/update.spec.ts

```diff
@@ -20,6 +21,39 @@ describe('modules/manager/maven/update', () => {
       expect(project.valueWithPath('version')).toBe('0.0.2');
     });
 
+    it('bumps pom.xml version keeping SNAPSHOT', () => {
+      const { bumpedContent } = pomUpdater.bumpPackageVersion(
+        minimumSnapshotContent,
+        '0.0.1-SNAPSHOT',
+        'patch'
+      );
+
+      const project = new XmlDocument(bumpedContent);
+      expect(project.valueWithPath('version')).toBe('0.0.2-SNAPSHOT');
+    });
+
+    it('bumps pom.xml minor version keeping SNAPSHOT', () => {
+      const { bumpedContent } = pomUpdater.bumpPackageVersion(
+        minimumSnapshotContent,
+        '0.0.1-SNAPSHOT',
+        'minor'
+      );
+
+      const project = new XmlDocument(bumpedContent);
+      expect(project.valueWithPath('version')).toBe('0.1.0-SNAPSHOT');
+    });
+
+    it('bumps pom.xml major version keeping SNAPSHOT', () => {
+      const { bumpedContent } = pomUpdater.bumpPackageVersion(
+        minimumSnapshotContent,
+        '0.0.1-SNAPSHOT',
+        'major'
+      );
+
+      const project = new XmlDocument(bumpedContent);
+      expect(project.valueWithPath('version')).toBe('1.0.0-SNAPSHOT');
+    });
```

Comment:
After playing around with this for a bit, I was surprised that it didn't work for our use case afterall. This is the behaviour that I was expecting:

```node
    it('returns content if bumping errors', () => {
      const { bumpedContent } = pomUpdater.bumpPackageVersion(
        simpleContent,
        '0.0.1',
        'prerelease'
      );

      const project = new XmlDocument(bumpedContent!);
      expect(project.valueWithPath('version')).toBe('0.0.2-SNAPSHOT');
    });
```

However, the `.0` was still present with this kind of bump. Is that an oversight or am I misunderstanding something?

Reply:
That's still https://github.com/npm/node-semver/issues/441 at work